pkg/optimizer/optimizer.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package optimizer
+
+import (
+	"errors"
+	"os"
+	"os/exec"
+)
+
+const (
+	MaxDataPathDepth = 4
```

Comment:
We have an env variable configuring the max path length.

Reply:
Do we want to use the same env variable? Currently its default is 2. Is that enough?

Reply:
Let's leave it for now and come back to it later after you implement optimization on different path lengths.